Applications/SpreadSheet/SpreadSheet.cpp

```diff
@@ -0,0 +1,294 @@
+    {
+        if (name.is_string()) {
+            if (auto pos = Sheet::parse_cell_name(name.as_string()); pos.has_value()) {
+                auto& cell = m_sheet.ensure(pos.value());
+                if (auto current = m_sheet.current_evaluated_cell())
+                    current->reference_from(&cell);
+
+                cell.set_data(value); // FIXME: This produces un-savable state!
+                return true;
+            }
+        }
+
+        return GlobalObject::put(name, value, receiver);
+    }
+
+    virtual void initialize() override
+    {
+        GlobalObject::initialize();
+        define_native_function("parse_cell_name", parse_cell_name, 1);
+    }
+
+    static JS_DEFINE_NATIVE_FUNCTION(parse_cell_name)
+    {
+        if (interpreter.argument_count() != 1)
+            return interpreter.throw_exception<JS::TypeError>("Expected exactly one argument to parse_cell_name()");
+
+        auto name_value = interpreter.argument(0);
+        if (!name_value.is_string())
+            return interpreter.throw_exception<JS::TypeError>("Expected a String argument to parse_cell_name()");
+
+        auto position = Sheet::parse_cell_name(name_value.as_string().string());
+        if (!position.has_value())
+            return JS::js_undefined();
+
+        auto object = JS::Object::create_empty(interpreter.global_object());
+        object->put("column", JS::js_string(interpreter, position.value().column));
+        object->put("row", JS::Value((unsigned)position.value().row));
+
+        return object;
+    }
+
+private:
+    virtual const char* class_name() const override { return "SheetGlobalObject"; }
+
+    Sheet& m_sheet;
+};
+
+Sheet::Sheet(const StringView& name)
+    : m_name(name)
+    , m_interpreter(JS::Interpreter::create<SheetGlobalObject>(*this))
+{
+    for (size_t i = 0; i < 20; ++i)
+        add_row();
+
+    for (size_t i = 0; i < 16; ++i)
+        add_column();
+
+    auto file_or_error = Core::File::open("/res/js/SpreadSheet/runtime.js", Core::IODevice::OpenMode::ReadOnly);
+    if (!file_or_error.is_error()) {
+        auto buffer = file_or_error.value()->read_all();
+        evaluate(buffer);
+    }
+}
+
+Sheet::~Sheet()
+{
+}
+
+size_t Sheet::add_row()
+{
+    return m_rows++;
+}
+
+String Sheet::add_column()
+{
+    if (m_current_column_name_length == 0) {
+        m_current_column_name_length = 1;
+        m_columns.append("A");
+        return "A";
+    }
+
+    if (m_current_column_name_length == 1) {
+        auto last_char = m_columns.last()[0];
+        if (last_char == 'Z') {
+            m_current_column_name_length = 2;
+            m_columns.append("AA");
+            return "AA";
+        }
+
+        last_char++;
+        m_columns.append({ &last_char, 1 });
+        return m_columns.last();
+    }
+
+    TODO();
+}
+
+void Sheet::update()
+{
+    m_visited_cells_in_update.clear();
+    for (auto& it : m_cells) {
```

Comment:
Since you have to repeatedly access `it.value` here, it might be nice to alias it:
```cpp
auto& cell = *it.value;
```